bootloader/test_infra/src/main.rs

```diff
@@ -0,0 +1,181 @@
+use tracing_subscriber::fmt;
+use tracing_subscriber::prelude::__tracing_subscriber_SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use vm::{
+    HistoryDisabled, L1BatchEnv, L2BlockEnv, SystemEnv, TxExecutionMode, Vm, VmExecutionMode,
+    VmTracer,
+};
+use zksync_contracts::{
+    read_zbin_bytecode, BaseSystemContracts, ContractLanguage, SystemContractCode,
+    SystemContractsRepo,
+};
+use zksync_state::{
+    InMemoryStorage, StoragePtr, StorageView, IN_MEMORY_STORAGE_DEFAULT_NETWORK_ID,
+};
+use zksync_types::system_contracts::get_system_smart_contracts_from_dir;
+use zksync_types::{block::legacy_miniblock_hash, Address, L1BatchNumber, MiniblockNumber, U256};
+use zksync_types::{L2ChainId, Transaction};
+use zksync_utils::bytecode::hash_bytecode;
+use zksync_utils::{bytes_to_be_words, u256_to_h256};
+
+mod hook;
+mod test_count_tracer;
+mod tracer;
+
+/// Executes the "internal transfer test" of the bootloader -- the test that
+/// returns the amount of gas needed to perform and internal transfer, assuming no gas price
+/// per pubdata, i.e. under assumption that the refund will not touch any new slots.
+fn execute_internal_bootloader_test() {
+    let test_location = env::current_dir()
+        .unwrap()
+        .join("../build/artifacts/bootloader_test.yul/bootloader_test.yul.zbin");
+    println!("Current dir is {:?}", test_location);
+    let bytecode = read_zbin_bytecode(test_location.as_path());
+    let hash = hash_bytecode(&bytecode);
+    let bootloader = SystemContractCode {
+        code: bytes_to_be_words(bytecode),
+        hash,
+    };
+
+    let repo = SystemContractsRepo {
+        root: env::current_dir().unwrap().join("../../"),
+    };
+
+    let bytecode = repo.read_sys_contract_bytecode("", "DefaultAccount", ContractLanguage::Sol);
+    let hash = hash_bytecode(&bytecode);
+    let default_aa = SystemContractCode {
+        code: bytes_to_be_words(bytecode),
+        hash,
+    };
+
+    let base_system_contract = BaseSystemContracts {
+        bootloader,
+        default_aa,
+    };
+
+    let system_env = SystemEnv {
+        zk_porter_available: false,
+        version: zksync_types::ProtocolVersionId::latest(),
+        base_system_smart_contracts: base_system_contract,
+        gas_limit: 80_000_000,
+        execution_mode: TxExecutionMode::VerifyExecute,
+        default_validation_computational_gas_limit: u32::MAX,
+        chain_id: zksync_types::L2ChainId(299),
+    };
+
+    let mut l1_batch_env = L1BatchEnv {
+        previous_batch_hash: None,
+        number: L1BatchNumber::from(1),
+        timestamp: 14,
+        l1_gas_price: 250_000_000,
+        fair_l2_gas_price: 250_000_000,
+        fee_account: Address::default(),
+
+        enforced_base_fee: None,
+        first_l2_block: L2BlockEnv {
+            number: 1,
+            timestamp: 15,
+            prev_block_hash: legacy_miniblock_hash(MiniblockNumber(0)),
+            max_virtual_blocks_to_create: 0,
```

Comment:
I understand that the test framework will never get to this point in code, but the first miniblock must create at least 1 virtual block